tests/auth/linking.node.test.ts

```diff
@@ -0,0 +1,186 @@
+    }
+
+    const linkDevice = async (data: Record<string, unknown>): Promise<void> => {
+      const { username, audience } = data as Record<string, string>
+
+      consumerAccounts[username] = consumerAccounts[username] ?? []
+      consumerAccounts[username] = [...consumerAccounts[username], audience]
+    }
+
+    setImplementations({
+      auth: {
+        ...LOCAL_IMPLEMENTATION.auth,
+        createChannel,
+        checkCapability,
+        delegateAccount,
+        linkDevice
+      }
+    })
+
+    // debug({ enabled: true })
+  })
+
+  afterEach(() => {
+    channel = new EventEmitter()
+    producerAccounts = {}
+    consumerAccounts = {}
+  })
+
+  it("links an account", async () => {
+    let consumerDone = false
+
+    const producer = await createProducer({ username: "elm-owl" })
+
+    producer.on("challenge", ({ confirmPin }) => {
+      confirmPin()
+    })
+
+    const consumer = await createConsumer({ username: "elm-owl" })
+
+    consumer.on("done", () => {
+      consumerDone = true
+    })
+
+    while (!consumerDone) await new Promise(r => setTimeout(r, 1000))
+    producer.cancel()
+
+    expect(consumerAccounts["elm-owl"]).toBeDefined()
+    expect(consumerAccounts["elm-owl"].length).toEqual(1)
+    expect(producerAccounts["elm-owl"]).toBeDefined()
+    expect(producerAccounts["elm-owl"].length).toEqual(1)
+    expect(consumerAccounts).toEqual(producerAccounts)
+  })
+
+  it("links when consumer starts first", async () => {
+    let consumerDone = false
+
+    const consumer = await createConsumer({ username: "elm-owl" })
+
+    consumer.on("done", () => {
+      consumerDone = true
+    })
+
+    const producer = await createProducer({ username: "elm-owl" })
+
+    producer.on("challenge", ({ confirmPin }) => {
+      confirmPin()
+    })
+
+    while (!consumerDone) await new Promise(r => setTimeout(r, 1000))
+    producer.cancel()
+
+    expect(consumerAccounts["elm-owl"]).toBeDefined()
+    expect(consumerAccounts["elm-owl"].length).toEqual(1)
+    expect(producerAccounts["elm-owl"]).toBeDefined()
+    expect(producerAccounts["elm-owl"].length).toEqual(1)
+    expect(consumerAccounts).toEqual(producerAccounts)
+  })
+
+  it("declines to link an account", async () => {
+    let consumerDone = false
+
+    const producer = await createProducer({ username: "elm-owl" })
+
+    producer.on("challenge", ({ rejectPin }) => {
+      rejectPin()
+    })
+
+    const consumer = await createConsumer({ username: "elm-owl" })
+
+    consumer.on("done", () => {
+      consumerDone = true
+    })
+
+    while (!consumerDone) await new Promise(r => setTimeout(r, 1000))
+    producer.cancel()
+
+    expect(consumerAccounts["elm-owl"]).not.toBeDefined()
+    expect(producerAccounts["elm-owl"]).not.toBeDefined()
+    expect(consumerAccounts).toEqual(producerAccounts)
+  })
+
+
+  // TODO: Run this test when we have implemented a message queue
+  it.skip("links with one producer and multiple consumers", async () => {
```

Comment:
We are skipping this test for now because one producer and multiple consumers does not work reliably. We should bring this test back when we implement a queue to track producer and consumer messages.

Reply:
Sounds fine. We can create an issue for that once this is merged I think :+1: